File: NDKbuild/app/src/main/java/ows/kotlinstudy/ndk_build/MainActivity.java
package ows.kotlinstudy.ndk_build;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    public static final String TAG = MainActivity.class.getSimpleName();

    static {
        System.loadLibrary("native-lib");
    }

    public native int plus(int num1, int num2);
    public native int minus(int num1, int num2);
    public native int multi(int num1, int num2);
    public native double div(int num1, int num2);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        int num1 = 10;
        int num2 = 20;
        Log.d(TAG, "num1 + num2 = "+plus(num1, num2));
        Log.d(TAG, "num1 - num2 = "+minus(num1, num2));
        Log.d(TAG, "num1 * num2 = "+multi(num1, num2));
        Log.d(TAG, "num1 / num2 = "+div(num1, num2));
    }
}

File: CalculatorWithMVVM/app/src/main/java/ows/kotlinstudy/calculatorwithmvvm/main/MainActivity.kt
package ows.kotlinstudy.calculatorwithmvvm.main

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.get
import androidx.recyclerview.widget.LinearLayoutManager
import ows.kotlinstudy.calculatorwithmvvm.R
import ows.kotlinstudy.calculatorwithmvvm.databinding.ActivityMainBinding
import ows.kotlinstudy.mvcexample.model.db.ResultDatabase

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.lifecycleOwner = this
        binding.viewmodel = MainViewModel(ResultDatabase.build(applicationContext))
        bindViews()
    }

    private fun bindViews(){
        binding.resultRecyclerView.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = ResultAdapter()
        }
    }
}